sharedbuffers/mapped_struct.py

```diff
@@ -1719,6 +1738,403 @@ def __str__(self):
+                    for i in xrange(xlen):
+                        val = self.objlist._c_getitem(i, dcode, objlen, itemsize, offset, _struct, None)
+                        if not pfset._contains_compressed(val):
+                            return False
+                    return not strict_subset or xlen < pfset.bitlen
+
+            elif self.objlist is None:
+                if self.bitrep_lo:
+                    for i in xrange(64):
+                        if not self.bitrep_lo >> i:
+                            break
+                        elif (self.bitrep_lo & (1 << i)) and i not in pfset:
+                            return False
+
+                if self.bitrep_hi:
+                    for i in xrange(64):
+                        if not self.bitrep_hi >> i:
+                            break
+                        elif (self.bitrep_hi & (1 << i)) and (i + 64) not in pfset:
+                            return False
+
+                return not strict_subset or xlen < len(pfset)
+
+            # Both proxies use sorted lists.
+            j = 0
+            seqlen = len(pfset.objlist)
+            if xlen > seqlen:
+                return False
+
+            dcode, objlen, itemsize, offset, _struct = self.objlist._metadata()
+            dcode2, objlen2, itemsize2, offset2, _struct2 = pfset.objlist._metadata()
+
+            if dcode2 in ('q', 'I', 'i', 'd') and cython.compiled:
+                # fast path, use the search_hkey variants
+                for i in xrange(xlen):
+                    val = self.objlist._c_getitem(i, dcode, objlen, itemsize, offset, _struct, None)
+                    tmp_idx = pfset._search_key(val, dcode, offset, j, seqlen - j, j, True)
+                    if tmp_idx >= seqlen - j:
+                        return False
+                    j = tmp_idx
+            else:
+                for i in xrange(xlen):
+                    val = self.objlist._c_getitem(i, dcode, objlen, itemsize, offset, _struct, None)
+                    h1 = _stable_hash(val)
+                    while True:
+                        val2 = pfset.objlist._c_getitem(j, dcode2, objlen2, itemsize2, offset2, _struct2, None)
+                        h2 = _stable_hash(val2)
+                        j += 1
+
+                        if h1 == h2 and val == val2:
+                            break
+                        elif h1 < h2 or j >= seqlen:
+                            return False
+                        else:
+                            # pfset[j] < val, skip as much as we can
+                            while j < seqlen:
+                                if _stable_hash(pfset.objlist._c_getitem(j, dcode2, objlen2, itemsize2, offset2, _struct2, None)) >= h1:
+                                    break
+                                j += 1
+                            if j == seqlen:
+                                return False
+            return not strict_subset or xlen < seqlen
+        elif xlen > len(seq):
```

Comment:
Why do you know that `seq` has the `len` attribute? For example, on Python 2, you could do the following:
```
>>> set([1,2,3]) < 3
False
```

Reply:
Ah, you're right. I'll fix that.